### ISSUE TITLE
No PropTypes validation

### ISSUE THREAD

**Issue by michelmattos** [keajs/kea#33]
I may be doing something wrong but I can't make the PropTypes validation works. If I have something like this:
```javascript
import React from 'react'
import PropTypes from 'prop-types'
import { kea } from 'kea'

const counterLogic = kea({
  actions: () => ({
    increment: amount => ({ amount }),
    decrement: amount => ({ amount })
  }),

  reducers: ({ actions }) => ({
    counter: [0, PropTypes.number, {
      [actions.increment]: (state, payload) => state + payload.amount,
      [actions.decrement]: (state, payload) => state - payload.amount
    }]
  })
})

const Counter = ({
  actions: {
    increment, decrement
  },
  counter
}) =>
  <div>
    <div>{ counter }</div>
    <button onClick={() => increment(1)}>+</button>
    <button onClick={() => decrement(1)}>-</button>
  </div>

export default counterLogic(Counter)
```

It works fine. But if I change the `increment` calling to something like this:
```javascript
    <button onClick={() => increment('abc')}>+</button>
```

It should raise an exception, right? Like it does with component proptypes validation.
But I'm not getting anything from console. If you run this component and try to `increment` something different from a number (like a `'abc'` string), your component will receive `'0abc'` from `counter` prop. Maybe I'm doing something wrong or maybe it's a bug. I don't know, lol

**Comment by mariusandra** [comment]
Hey, thanks for raising this issue! Indeed this doesn't work :(.

Kea doesn't check the `propTypes` itself, it just adds the right ones to the wrapped component's ones and delegates the checking to React. Unfortunately React doesn't support propTypes for functional components, so nothing is checked. If you would rewrite your `Counter` as an ES class, everything would work... but that's not the right solution.

Unfortunately I'm not sure what is the right solution. I could wrap functional components with another function that manually checks the propTypes in development mode. 

I'll check this out and report if it's feasible.

**Comment by michelmattos** [comment]
Actually, you can add proptypes checking to functional component with this:
```javascript
Counter.propTypes = {
  counter: PropTypes.number
}
```
Adding this will raise a warning if you click the `increment` button. I'll check if there is a solution for this too.

**Comment by michelmattos** [comment]
Maybe we can use `PropTypes.checkPropTypes`? What do you think, @mariusandra ?

**Comment by michelmattos** [comment]
Anyway, I'll look into this and see if I can come up with something (maybe a PR). Congratulations for your daughter! :)

**Comment by mariusandra** [comment]
Hey, you're right! Functional components do support propTypes! There was just a bug in the code, that's now fixed with `0.23.2`. Please check it out and let me know if it works :).

**Comment by michelmattos** [comment]
Yay! It's working now! Thanks, @mariusandra!
